src/discussions/comments/CommentsView.jsx

```diff
@@ -77,47 +77,66 @@ function DiscussionCommentsView({
+  const unEndorsedComments = useMemo(() => [...filterPosts(comments, 'unendorsed')], [comments]);
 
-      <div className="mx-4" role="list">
-        {sortedComments.map(comment => (
-          <Comment comment={comment} key={comment.id} postType={postType} isClosedPost={isClosed} />
-        ))}
-        {hasMorePages && !isLoading && (
-          <Button
-            onClick={handleLoadMoreResponses}
-            variant="link"
-            block="true"
-            className="card p-4 mb-4 font-weight-500 font-size-14"
-            style={{
-              lineHeight: '20px',
-            }}
-            data-testid="load-more-comments"
-          >
-            {intl.formatMessage(messages.loadMoreResponses)}
-          </Button>
-        )}
-        {isLoading
+  const handleDefinition = (message, commentsLength) => (
+    <div className="mx-4 text-primary-700" role="heading" aria-level="2" style={{ lineHeight: '28px' }}>
+      {intl.formatMessage(message, { num: commentsLength })}
+    </div>
+  );
+
+  const handleComments = (postComments, post, isClosedPost, showAddResponse = false) => (
+    <div className="mx-4" role="list">
+      {postComments.map(comment => (
+        <Comment comment={comment} key={comment.id} postType={post} isClosedPost={isClosedPost} />
+      ))}
+      {hasMorePages && !isLoading && (
+        <Button
+          onClick={handleLoadMoreResponses}
+          variant="link"
+          block="true"
+          className="card p-4 mb-4 font-weight-500 font-size-14"
+          style={{
+            lineHeight: '20px',
+          }}
+          data-testid="load-more-comments"
+        >
+          {intl.formatMessage(messages.loadMoreResponses)}
+        </Button>
+      )}
+      {isLoading
         && (
           <div className="card my-4 p-4 d-flex align-items-center">
             <Spinner animation="border" variant="primary" />
           </div>
         )}
-        {!!sortedComments.length && !isClosed
-          && <ResponseEditor postId={postId} addWrappingDiv />}
-      </div>
+      {!!(endorsedComments.length + unEndorsedComments.length) && !isClosed && showAddResponse
+        && <ResponseEditor postId={postId} addWrappingDiv />}
+    </div>
+  );
+  return (
+    <>
+      {((hasMorePages && isLoading) || !isLoading)
+        && (
+          <>
+            {endorsedComments.length > 0
+              && (
+                <>
+                  {handleDefinition(messages.endorsedResponseCount, endorsedComments.length)}
+                  {handleComments(endorsedComments, postType, isClosed)}
+                </>
+              )}
+            {endorsed !== EndorsementStatus.ENDORSED
+              && (
+                <>
+                  {handleDefinition(messages.responseCount, unEndorsedComments.length)}
+                  {handleComments(unEndorsedComments, postType, isClosed, true)}
+                </>
+              )}
+          </>
+        )}
```

Comment:
Please avoid the extra lines for `&&`
{((hasMorePages && isLoading) || !isLoading) && (